juju/juju
# Netplan bridger.

## Description of change
Netplan will be default network configuration mechanism in 17.10,
replacing /etc/network/interfaces.
If /e/n/i exists we use it to configure bridges on hosts for containers,
if it doesn't we use netplan.

The code loads all netplan yamls from /etc/netplan, adds the requested
bridges and writes the whole netplan as one file while backing up the
original files. Interfaces to bridge are detected either by MAC address
or by name.

There is a known bug with interfaces defined in multiple files - those
are not supported yet.

## QA steps
1. Create a machine with 17.10 (netplan as a default) or 17.04 and convert its networking to netplan 
2. Create a container on this machine.
3. Verify that the bridges were properly created and that /etc/netplan/ contains both newly written 'total'
yaml and backups of old files.

## Documentation changes

## Bug reference
